### ISSUE TITLE
Extend ScatterElements op with min and max reduction 

### ISSUE THREAD

**Issue by samvanstroud** [onnx/onnx#4322]
Following https://github.com/onnx/onnx/pull/3484, the ScatterElements operation supports `add` and `mul` reduction operations. Additionally supporting `min`, `max`, and `mean` operations would allow for the remaining unsupported [`torch_scatter`](https://github.com/rusty1s/pytorch_scatter) ops to be supported by ONNX. 

I think this would allow [PyG](https://github.com/pyg-team/pytorch_geometric) models to be exported to ONNX. Currently the PyG [`softmax`](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/utils/softmax.py#L43) implementation uses `scatter_max` and `scatter_sum`. I'm hoping that these changes would be fairly easy to implement since the other reduction operations are already supported.

Tagging @shubhambhokare1 who contributed https://github.com/onnx/onnx/pull/3484 for guidance.

**Comment by samvanstroud** [comment]
I found that torch now has it's own `scatter_reduce`, so we probably could probably just add support for that and forget about the `torch_scatter` package.

**Comment by ramkrishna2910** [comment]
This is a work in progress right now.
Quiet a few Graph neural networks also need this functionality.

**Comment by samvanstroud** [comment]
Hi @ramkrishna2910, thanks. Do you know who is working on this and when it might be ready?

**Comment by ramkrishna2910** [comment]
@philass and I are looking in to this right now

**Comment by philass** [comment]
I have started a PR for it locally here, https://github.com/philass/onnx/pull/1

I will submit it to ONNX when it is more polished

**Comment by philass** [comment]
Yes it's been available since opset 18.

If you want it working `torch_scatter`, you can look at this [file](https://github.com/philass/onnx-experiments/blob/master/export-custom.py) for insipration
